README.rst

```diff
@@ -429,3 +429,80 @@ Sample test execution logs
    INFO    - Writing xunit report /home/arm/projects/zephyrproject/zephyr/twister-out/twister.xml...
    INFO    - Writing xunit report /home/arm/projects/zephyrproject/zephyr/twister-out/twister_report.xml...
    INFO    - Run completed
+
+Adding an external WIFI interface shield based on MikroE click boards
+**********************************************************************
+
+It is possible to add an external WiFI interface based on MikroElektronika click boards attaching
+them to the arduino_serial or mikrobus_serial nodes of a board like B-U585I-IOT02A, these shields are based
+on ESP8266 which are very affordable way to run the samples on SoCs that don't have a native networking support.
+
+.. image:: https://github.com/Linaro/zephyr_secure_inference/blob/main/docs/wifi-esp-click.png?raw=true
+  :alt: MikroElektronika WiFi-ESP-Click board.
+
+The board pinout is referred below:
+
++-----------------------+---------------------------------------------+
+| Shield Connector Pin  | Function                                    |
++=======================+=============================================+
+| RST#                  | ESP8266 Module Reset                        |
++-----------------------+---------------------------------------------+
+| TXD                   | Serial data transmission output pin         |
++-----------------------+---------------------------------------------+
+| RXD                   | Serial data reception input pin             |
++-----------------------+---------------------------------------------+
+
+Before usage, This shield should be loaded with the `ESP8266 AT Bin`_ software which is
+available at Espressif Systems web site. This version is command
+compatible with ESP8266 AT Bin 2.2.0 or 1.7.0, after getting the binary from Espressif
+site, connect the J1 of the board to a serial-to-USB converter of your
+preference, or to one Espressif programming boards like ESP-Prog on the HD1 connector,
+short circuit pins 5 and 6 to put the ESP8266 into download mode, on other modules, 
+this is the same to tie the IO0 pin to GND. Install the `ESP-Tool`_, extract the folder,
+navigate inside the binaries location and type the following command to flash the device:
+
+.. code-block:: console
+
+   esptool.py --chip auto \
+   --before default_reset \
```

Comment:
What version of `esptool` are you using here?

When I test this with 3.3 I get:

```bash
$ esptool \   
  --port /dev/tty.usbserial-21110 \
  --chip esp8266 \
  --baud 115200
  --before default_reset \
  --after hard_reset write_flash \
  --flash_mode dio \
  --flash_freq 80m \
  --flash_size 2MB \
  0x8000 partition_table/partition-table.bin \
  0x9000 ota_data_initial.bin \
  0x0 bootloader/bootloader.bin \
  0x10000 esp-at.bin \
  0xF0000 at_customize.bin \
  0xFC000 customized_partitions/client_ca.bin \
  0x106000 customized_partitions/mqtt_key.bin \
  0x104000 customized_partitions/mqtt_cert.bin \
  0x108000 customized_partitions/mqtt_ca.bin \
  0xF1000 customized_partitions/factory_param.bin \
  0xF8000 customized_partitions/client_cert.bin \
  0xFA000 customized_partitions/client_key.bin
esptool.py v3.3
usage: esptool [-h]
               [--chip {auto,esp8266,esp32,esp32s2,esp32s3beta2,esp32s3,esp32c3,esp32c6beta,esp32h2beta1,esp32h2beta2,esp32c2}]
               [--port PORT] [--baud BAUD]
               [--before {default_reset,usb_reset,no_reset,no_reset_no_sync}]
               [--after {hard_reset,soft_reset,no_reset,no_reset_stub}] [--no-stub] [--trace]
               [--override-vddsdio [{1.8V,1.9V,OFF}]] [--connect-attempts CONNECT_ATTEMPTS]
               {load_ram,dump_mem,read_mem,write_mem,write_flash,run,image_info,make_image,elf2image,read_mac,chip_id,flash_id,read_flash_status,write_flash_status,read_flash,verify_flash,erase_flash,erase_region,merge_bin,get_security_info,version}
               ...

esptool.py v3.3 - Espressif chips ROM Bootloader Utility

positional arguments:
  {load_ram,dump_mem,read_mem,write_mem,write_flash,run,image_info,make_image,elf2image,read_mac,chip_id,flash_id,read_flash_status,write_flash_status,read_flash,verify_flash,erase_flash,erase_region,merge_bin,get_security_info,version}
                        Run esptool {command} -h for additional help
    load_ram            Download an image to RAM and execute
    dump_mem            Dump arbitrary memory to disk
    read_mem            Read arbitrary memory location
    write_mem           Read-modify-write to arbitrary memory location
    write_flash         Write a binary blob to flash
    run                 Run application code in flash
    image_info          Dump headers from an application image
    make_image          Create an application image from binary files
    elf2image           Create an application image from ELF file
    read_mac            Read MAC address from OTP ROM
    chip_id             Read Chip ID from OTP ROM
    flash_id            Read SPI flash manufacturer and device ID
    read_flash_status   Read SPI flash status register
    write_flash_status  Write SPI flash status register
    read_flash          Read SPI flash content
    verify_flash        Verify a binary blob against flash
    erase_flash         Perform Chip Erase on SPI flash
    erase_region        Erase a region of the flash
    merge_bin           Merge multiple raw binary files into a single file for later flashing
    get_security_info   Get some security-related data
    version             Print esptool version

optional arguments:
  -h, --help            show this help message and exit
  --chip {auto,esp8266,esp32,esp32s2,esp32s3beta2,esp32s3,esp32c3,esp32c6beta,esp32h2beta1,esp32h2beta2,esp32c2}, -c {auto,esp8266,esp32,esp32s2,esp32s3beta2,esp32s3,esp32c3,esp32c6beta,esp32h2beta1,esp32h2beta2,esp32c2}
                        Target chip type
  --port PORT, -p PORT  Serial port device
  --baud BAUD, -b BAUD  Serial port baud rate used when flashing/reading
  --before {default_reset,usb_reset,no_reset,no_reset_no_sync}
                        What to do before connecting to the chip
  --after {hard_reset,soft_reset,no_reset,no_reset_stub}, -a {hard_reset,soft_reset,no_reset,no_reset_stub}
                        What to do after esptool.py is finished
  --no-stub             Disable launching the flasher stub, only talk to ROM bootloader. Some
                        features will not be available.
  --trace, -t           Enable trace-level output of esptool.py interactions.
  --override-vddsdio [{1.8V,1.9V,OFF}]
                        Override ESP32 VDDSDIO internal voltage regulator (use with care)
  --connect-attempts CONNECT_ATTEMPTS
                        Number of attempts to connect, negative or 0 for infinite. Default: 7.
zsh: command not found: --before
```

Reply:
Removing `--before` seems to resolve this, enabling updates.

Reply:
There is a bug on the command line of esptool, the port or baud options must be present before the --before option.